### PR TITLE
[generator] Enable warnaserror for our bindings

### PR DESCRIPTION
**Pull request by dalexsoto**
Enabled `warnaserror` in our SDK bindings

### CODE REVIEW

**Comment by dalexsoto** [comment]
@spouliot Fixed and thanks, great catch

File: src/Makefile [hunk 1]
```diff
@@ -22,6 +22,16 @@ include $(TOP)/opentk/Makefile.include
 include $(TOP)/tools/pmcs/Makefile.frag
 
 #
+# Specific warnings that we want reported as errors by generator
+# comma separated list
+#
+GENERATOR_WARNASERROR=
```
Comment (rolfbjarne) [inline src/Makefile]: Please add a comment that an empty list means reporting all warnings as errors.

File: src/Makefile [hunk 2]
```diff
@@ -525,6 +535,7 @@ $(MAC_BUILD_DIR)/$(1)/generated-sources: $$(MAC_$(1)_GENERATOR) $(MAC_APIS) $(MA
 		-d:XAMARIN_MAC \
 		-compiler:$(MAC_BUILD_DIR)/$(1)/pmcs \
 		-process-enums \
+		-warnaserror:$(MAC_GENERATOR_WARNASERROR) \
```
Comment (rolfbjarne) [inline src/Makefile]: The code that generates generator-ikvm.csproj needs to be updated as well.

**Comment by dalexsoto** [comment]
@rolfbjarne fixed, thanks!

**Pull request merged**